tests/test_templates_broadcast.py

```diff
@@ -0,0 +1,267 @@
+    def test_prepares_correct_state(self, structure, parameters, unitary, target):
+        """Tests the state produced by different unitaries."""
+
+        @qml.qnode(dev_4_qubits)
+        def circuit():
+            for w in range(4):
+                qml.PauliX(wires=w)
+            broadcast(block=unitary, structure=structure, wires=range(4), parameters=parameters)
+            return [qml.expval(qml.PauliZ(wires=w)) for w in range(4)]
+
+        res = circuit()
+        assert np.allclose(res, target)
+
+    @pytest.mark.parametrize("structure, parameters, unitary, target, observable", CV_TARGET_OUTPUTS)
+    def test_prepares_correct_state_cv(self, structure, parameters, unitary, target, observable):
+        """Tests the state produced by different unitaries."""
+
+        @qml.qnode(dev_4_qumodes)
+        def circuit():
+            for w in range(4):
+                Displacement(1, 0, wires=w)
+            broadcast(block=unitary, structure=structure, wires=range(4), parameters=parameters)
+            return [qml.expval(observable(wires=w)) for w in range(4)]
+
+        res = circuit()
+        assert np.allclose(res, target)
+
+    @pytest.mark.parametrize("parameters, n_wires", [(np.array([0]), 2),
+                                                     ([0, 0, 0, 1, 0], 3)])
+    def test_throws_error_when_mismatch_params_wires(self, parameters, n_wires):
+        """Tests that error thrown when 'parameters' does not contain one set
+           of parameters for each wire."""
+
+        dev = qml.device('default.qubit', wires=n_wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            broadcast(block=RX, wires=range(n_wires), parameters=parameters)
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match="'parameters' must contain entries for"):
+            circuit()
+
+    def test_exception_wires_not_valid(self):
+        """Tests that an exception is raised if 'wires' argument has invalid format."""
+
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            broadcast(block=RX, wires='a', parameters=[[1], [2]])
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match="wires must be a positive"):
+            circuit()
+
+    def test_exception_parameters_not_valid(self):
+        """Tests that an exception is raised if 'parameters' argument has invalid format."""
+
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            broadcast(block=RX, wires=[0, 1], parameters=RX)
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match="'parameters' must be either of type None or "):
+            circuit()
+
+    def test_exception_default_case_not_valid(self):
+        """Tests that an exception is raised if 'structure=None' for a 'block' acting on more than
```

Comment:
This test could be modified such that there is a difference between the template that was specified and the `structure` passed, that is e.g. `structure='double'`, but `block=ThreeWireTemplate`.

Reply:
Sorry, I left halfway in between editing this PR and hadn't updated the tests yet. 

For now I just delete the test, since we go with "no default". Again, for this to work elegantly we'd need a wire checking function for templates.